Source/Csla/DataPortalClient/LocalProxy.cs

```diff
@@ -30,19 +30,19 @@ public LocalProxy(ApplicationContext applicationContext, LocalProxyOptions optio
       ApplicationContext = applicationContext;
       Options = options;
 
-      if (Options.CreateScopePerCall)
+      if (Options.CreateScopePerCall && 
+        ApplicationContext.LogicalExecutionLocation == ApplicationContext.LogicalExecutionLocations.Client)
       {
         // create new DI scope and provider
         _scope = ApplicationContext.CurrentServiceProvider.CreateScope();
         var provider = _scope.ServiceProvider;
 
         // create and initialize new "server-side" ApplicationContext and manager
         var parentContext = applicationContext;
-        var ctx = new Csla.Core.ApplicationContextManager();
+        var ctx = provider.GetRequiredService<Core.IContextManager>();
```

Comment:
I wonder if the ctx variable would be better called mgr; I think this is the context manager and not the context at this point. This makes no difference to execution, only to whether there is a chance for greater understanding for those who come after us.